gitlab-sso.md

```diff
@@ -0,0 +1,19 @@
+# Configuring GitLab Single-Sign-On
+
+The following steps can be used to configure Mattermost to use GitLab as a single-sign-on (SSO) service.
+
+1. Login to your GitLab account and go to the Applications section either in Profile Settings or Admin Area.
+2. Add a new application called "Mattermost" with the following as Redirect URIs:
+  * `<your-mattermost-url>/login/gitlab/complete` (example: http://localhost:8065/login/gitlab/complete)
+  * `<your-mattermost-url>/signup/gitlab/complete`
+
+3. Submit the application and copy the given _Id_ and _Secret_ into the appropriate _SSOSettings_ fields in config/config.json
+
+4. Also in config/config.json, set _Allow_ to `true` for the _gitlab_ section, leave _Scope_ blank and use the following for the endpoints:
```

Comment:
Hi @maxlazio , `Scope` is defined in the OAuth2 spec here https://tools.ietf.org/html/rfc6749#section-3.3 . GitLab doesn't use `Scope` in it's OAuth2 service provider implementation, so it is left blank here. The reason we have it at all is for other systems we would like to use as OAuth 2 providers. For example, Google requires specifying a `Scope` for the access token that will control how much the requesting app gets access too.

Does that make sense?